src/main/java/uk/gov/hmcts/reform/jps/repository/StatusHistoryRepository.java

```diff
@@ -5,16 +5,35 @@
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 import uk.gov.hmcts.reform.jps.domain.StatusHistory;
+import uk.gov.hmcts.reform.jps.model.RecordingUser;
 
+import java.time.LocalDate;
 import java.util.List;
 
 @Repository
-public interface StatusHistoryRepository extends JpaRepository<StatusHistory, Long> {
+public interface StatusHistoryRepository extends JpaRepository<StatusHistory, Long>, SittingRecordRepositorySearch {
 
     @Query("select sr from StatusHistory sr where sr.sittingRecord.id=:id ORDER BY sr.id asc")
     List<StatusHistory> findStatusHistoryAsc(@Param("id") Long id);
 
     @Query("select sr from StatusHistory sr where sr.sittingRecord.id=:id ORDER BY sr.id desc")
     List<StatusHistory> findStatusHistoryDesc(@Param("id") Long id);
 
+
+    @Query("select distinct new uk.gov.hmcts.reform.jps.model.RecordingUser(sh.changeByUserId, sh.changeByName) "
+        + "from StatusHistory sh inner join SittingRecord sr "
+        + "on sh.sittingRecord.id = sr.id and sh.statusId = 'RECORDED' "
+        + "where sr.hmctsServiceId = :hmctsServiceId "
+        + "and ( CAST(:regionId as org.hibernate.type.StringType) is null "
```

Comment:
Is casting required?

Reply:
It's needed. Was introduced because of errors such as
org.postgresql.util.PSQLException: ERROR: could not determine data type of parameter $2